core/src/main/scala/org/apache/spark/util/logging/FileAppender.scala

```diff
@@ -76,7 +80,13 @@ private[spark] class FileAppender(inputStream: InputStream, file: File, bufferSi
           }
         }
       } {
-        closeFile()
+        try {
+          if (closeStreams) {
+            inputStream.close()
+          }
```

Comment:
This's the normal exit path. So, ideally, the executor would also stop logging afterward, right? But, of course, it's safer to close it.

And I think we should also handle the error path below. That one looks more critical.

Reply:
@jhu-chang what do you think of this last comment?

Reply:
@Ngone51 @srowen 
It's for both normal exit and exception.
Sorry, i don't quite understand the last comment: do you mean handling the error from "inputStream.close()"?

Reply:
Oh, I see. It's inside the finally block so should work for both normal exit and exception.